sdk/client-core/core/src/main/java/io/clientcore/core/http/client/DefaultHttpClient.java

```diff
@@ -239,12 +244,12 @@ private Response<?> receiveResponse(HttpRequest httpRequest, HttpURLConnection c
                     responseBodyMode = BUFFER;
                 }
 
-                httpRequest.getRequestOptions().setResponseBodyMode(responseBodyMode); // We only change this if it was null.
+                requestOptions.setResponseBodyMode(responseBodyMode); // We only change this if it was null.
```

Comment:
*hand waving architect suggestion*
We should never mutate the `RequestOptions` values in our code - we should only ever read them